### PR TITLE
Deprecate RedshiftDataOperatorAsync, RedshiftClusterSensorAsync, S3KeySensorAsync, BatchOperatorAsync, SageMakerProcessingOperatorAsync, SageMakerTransformOperatorAsync

### PR DESCRIPTION
**Pull request by Lee-W**
Deprecate `RedshiftDataOperatorAsync`, `RedshiftClusterSensorAsync`, `S3KeySensorAsync`, `BatchOperatorAsync`, `SageMakerProcessingOperatorAsync`, `SageMakerTransformOperatorAsync` and proxy them to their Airflow OSS provider's counterpart

### CODE REVIEW

**Comment by Lee-W** [comment]
image

**Comment by Lee-W** [comment]
S3
image

**Comment by Lee-W** [comment]
SageMakerProcessingOperatorAsync, SageMakerTransformOperatorAsync

image

**Comment by Lee-W** [comment]
RedshiftClusterSensorAsync

image

File: astronomer/providers/amazon/aws/operators/redshift_data.py [hunk 1]
```diff
@@ -29,47 +17,14 @@ def __init__(
...
-        """
-        if event:
-            if "status" in event and event["status"] == "error":
-                msg = "context: {}, error message: {}".format(context, event["message"])
-                raise AirflowException(msg)
-            elif "status" in event and event["status"] == "success":
-                self.log.info("%s completed successfully.", self.task_id)
-        else:
-            raise AirflowException("Did not receive valid event from the trigerrer")
+        kwargs["poll_interval"] = poll_interval
```
Comment (pankajastro) [inline astronomer/providers/amazon/aws/operators/redshift_data.py:29]: do we need this line? 
looks like `poll_interval ` is kwargs we can just remove this from the init

File: astronomer/providers/amazon/aws/sensors/redshift_cluster.py [hunk 2]
```diff
@@ -27,45 +19,20 @@ def __init__(
     ):
         # TODO: Remove once deprecated
         if poll_interval:
-            self.poke_interval = poll_interval
+            kwargs["poke_interval"] = poll_interval
```
Comment (pankajastro) [inline astronomer/providers/amazon/aws/sensors/redshift_cluster.py:22]: shall we remove `poll_interval` from init and check kwargs["poll_interval"] in L21 this way code editor will probably not suggest this param

```python
if kwargs.get("poll_interval"):
    kwargs["poke_interval"] = kwargs["poll_interval"]
```

File: setup.cfg [hunk 3]
```diff
@@ -118,7 +118,7 @@ mypy =
 # All extras from above except 'mypy', 'docs' and 'tests'
 all =
     aiobotocore>=2.1.1
-    apache-airflow-providers-amazon>=8.16.0
+    apache-airflow-providers-amazon>=8.17.0rc1
```
Comment (pankajastro) [inline setup.cfg:121]: Note: please remove rc before merging it

File: astronomer/providers/amazon/aws/sensors/redshift_cluster.py (hunk 2 above)
Comment (Lee-W) [reply astronomer/providers/amazon/aws/sensors/redshift_cluster.py]: good idea! just changed

File: astronomer/providers/amazon/aws/hooks/sagemaker.py [hunk 4]
```diff
@@ -21,56 +24,67 @@ class SageMakerHookAsync(AwsBaseHookAsync):
 
     Additional arguments (such as ``aws_conn_id``) may be specified and
     are passed down to the underlying AwsBaseHookAsync.
+
+    This class is deprecated and will be removed in 2.0.0.
+    Use :class: `~airflow.providers.amazon.aws.sensors.sagemaker.SageMakerHook` instead
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/hooks/sagemaker.py:29]:
```suggestion
    Use :class: `~airflow.providers.amazon.aws.hooks.sagemaker.SageMakerHook` instead
```

File: astronomer/providers/amazon/aws/hooks/sagemaker.py [hunk 5]
```diff
@@ -21,56 +24,67 @@ class SageMakerHookAsync(AwsBaseHookAsync):
...
     """
 
     NON_TERMINAL_STATES = ("InProgress", "Stopping", "Stopped")
 
     def __init__(self, *args: Any, **kwargs: Any):
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
+                "Please use `airflow.providers.amazon.aws.sensors.sagemaker.SageMakerHook`"
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/hooks/sagemaker.py:38]:
```suggestion
                "Please use `airflow.providers.amazon.aws.hooks.sagemaker.SageMakerHook`"
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 6]
```diff
@@ -31,216 +31,42 @@ def serialize(result: dict[str, Any]) -> str:
...
-        that this operation waits to check the status of the transform job.
-    :param max_ingestion_time: The operation fails if the transform job doesn't finish
-     within max_ingestion_time seconds. If you set this parameter to None, the operation does not timeout.
-    :param check_if_job_exists: If set to true, then the operator will check whether a transform job
-        already exists for the name in the config.
-    :param action_if_job_exists: Behaviour if the job name already exists. Possible options are "increment"
-        (default) and "fail".
-        This is only relevant if check_if_job_exists is True.
+    This class is deprecated.
+    Please use :class: `~airflow.providers.amazon.aws.operators.sagemaker.SageMakerProcessingOperator`
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:55]:
```suggestion
    Please use :class: `~airflow.providers.amazon.aws.operators.sagemaker.SageMakerTransformOperator`
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 7]
```diff
@@ -31,216 +31,42 @@ def serialize(result: dict[str, Any]) -> str:
...
-                end_time=end_time,
-                aws_conn_id=self.aws_conn_id,
-                job_name=transform_config["TransformJobName"],
-                job_type="Transform",
-                response_key="TransformJobStatus",
+    def __init__(self, **kwargs: Any) -> None:
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
+                "Please use `airflow.providers.amazon.aws.operators.sagemaker.SageMakerProcessingOperator`"
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:63]:
```suggestion
                "Please use `airflow.providers.amazon.aws.operators.sagemaker.SageMakerTransformOperator`"
```

File: astronomer/providers/amazon/aws/sensors/redshift_cluster.py [hunk 8]
```diff
@@ -1,71 +1,36 @@
...
-                self.log.info("%s completed successfully.", self.task_id)
-                self.log.info(
-                    "Cluster Identifier %s is in %s state", self.cluster_identifier, self.target_status
-                )
-                return None
-        self.log.info("%s completed successfully.", self.task_id)
-        return None
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/sensors/redshift_cluster.py:29]:
```suggestion
                "This module is deprecated."
```
I guess we won't remove the sensors and operators?

File: astronomer/providers/amazon/aws/operators/batch.py [hunk 9]
```diff
@@ -7,95 +7,29 @@
...
-        Relies on trigger to throw an exception, otherwise it assumes execution was
-        successful.
-        """
-        if "status" in event and event["status"] == "error":
-            raise AirflowException(event["message"])
-        self.log.info(event["message"])
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/batch.py:28]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/operators/redshift_data.py [hunk 10]
```diff
@@ -1,75 +1,27 @@
...
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=RedshiftDataTrigger(
-                task_id=self.task_id,
-                poll_interval=self.poll_interval,
-                aws_conn_id=self.aws_conn_id,
-                query_ids=query_ids,
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/redshift_data.py:20]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 11]
```diff
@@ -31,216 +31,42 @@ def serialize(result: dict[str, Any]) -> str:
...
-            timeout=self.execution_timeout,
-            trigger=SagemakerProcessingTrigger(
-                poll_interval=self.check_interval,
-                aws_conn_id=self.aws_conn_id,
-                job_name=self.config["ProcessingJobName"],
-                end_time=end_time,
+    def __init__(self, **kwargs: Any) -> None:
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:42]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py [hunk 12]
```diff
@@ -31,216 +31,42 @@ def serialize(result: dict[str, Any]) -> str:
...
-                poke_interval=self.check_interval,
-                end_time=end_time,
-                aws_conn_id=self.aws_conn_id,
-                job_name=transform_config["TransformJobName"],
-                job_type="Transform",
-                response_key="TransformJobStatus",
+    def __init__(self, **kwargs: Any) -> None:
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:62]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/sensors/s3.py [hunk 13]
```diff
@@ -1,131 +1,30 @@
...
-                use_regex=self.use_regex,
-                aws_conn_id=self.aws_conn_id,
-                verify=self.verify,
-                poke_interval=self.poke_interval,
-                soft_fail=self.soft_fail,
-                should_check_fn=self.should_check_fn,
+    def __init__(self, **kwargs: Any) -> None:
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/sensors/s3.py:20]:
```suggestion
                "This module is deprecated."
```

**Review by pankajkoti — APPROVED** [review]
LGTM once above review comments are addressed.

**Pull request merged**